File: client/src/app/modules/quotation/services/shal.notice.template.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/appRest.service';
import { ShalNoticeTemplate } from '../../quotation/model/ShalNoticeTemplate';

@Injectable({
  providedIn: 'root'
})
export class ShalNoticeTemplateService extends AppRestService<ShalNoticeTemplate>{

  constructor(http: HttpClient) {
    super(http, "quotation");
  }

  getShalNoticeTemplates() {
    return this.getList(new HttpParams(), "shal-notice-templates");
  }
  
   addOrUpdateShalNoticeTemplate(shalNoticeTemplate: ShalNoticeTemplate) {
    return this.addOrUpdate(new HttpParams(), "shal-notice-template", shalNoticeTemplate, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/quotation/service/ShalNoticeTemplateService.java
package com.jss.osiris.modules.quotation.service;

import java.util.List;

import com.jss.osiris.modules.quotation.model.ShalNoticeTemplate;

public interface ShalNoticeTemplateService {
    public List<ShalNoticeTemplate> getShalNoticeTemplates();

    public ShalNoticeTemplate getShalNoticeTemplate(Integer id);
	
	 public ShalNoticeTemplate addOrUpdateShalNoticeTemplate(ShalNoticeTemplate shalNoticeTemplate);
}
